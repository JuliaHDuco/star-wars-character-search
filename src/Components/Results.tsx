import * as React from 'react';
import { IPerson } from '../types';
import Result from './Result';
import './styling/Results.scss'

interface IResultProps {
  results: IPerson[];
}

const Results = ({results}: IResultProps) => {
  if (results.length <= 0) {
    return <div>Searching...</div>;
  };

  return (
    <ul>
      {results.map((el: IPerson) => <Result key={el.name.toLowerCase().replace(' ', '_')} content={el}/>)}
    </ul>
  )
}

export default Results;
