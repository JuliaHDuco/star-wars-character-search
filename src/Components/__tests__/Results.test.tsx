import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { testResults } from '../../utilities/tests/data/results';
import Results from '../Results';

test('should match snapshot', () => {
  const component = renderer.create(
    <Results results={testResults} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render loader if no results', () => {
  const component = renderer.create(
    <Results results={[]} />
  );
  const loader = component.root.findAllByType('div')[0];
  
  expect(loader.props.children).toBe('Searching...');
});
