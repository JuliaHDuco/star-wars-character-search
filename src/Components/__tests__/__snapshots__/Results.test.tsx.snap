// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Luke Skywalker
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Owen Lars
    </button>
  </li>
</ul>
`;
